### PR TITLE
NMS-9224: Added Grafana dashlet

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-9224

### CODE REVIEW

File: features/vaadin-dashlets/dashlet-grafana/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/GrafanaDashletConfigurationWindow.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+                final Map<String, String> resultSet = new TreeMap<>();
+
+                try (CloseableHttpResponse httpResponse = httpClient.execute(httpGet);) {
+                    HttpEntity httpEntity = httpResponse.getEntity();
+                    if (httpEntity != null) {
+                        /**
+                         * Fill the result set...
+                         */
+                        final String responseString = IOUtils.toString(httpEntity.getContent(), "UTF-8");
```
Comment (j-white) [inline features/vaadin-dashlets/dashlet-grafana/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/GrafanaDashletConfigurationWindow.java]: Hmmm, I guess our Checkstyle rule doesn't work: https://github.com/OpenNMS/opennms/blob/foundation-2017/checkstyle/src/main/resources/nms_checks.xml#L20

**Comment by j-white** [comment]
I tested this out and it works as described. Having to login to Grafana isn't ideal, but increasing the session timeout can help reduce the frequency at which you'll need to login :smile:

**Comment by christianpape** [comment]
If the Grafana stuff isn't critical concerning security you can also allow anonymous auth and assign the viewer role for a given organization to these anonymous users. This way you don't have to be signed in for viewing dashboards.

**Comment by indigo423** [comment]
We have for each dashlet a documentation section: http://docs.opennms.org/opennms/branches/foundation-2017/guide-admin/guide-admin.html#_dashlets
This should be added for this dashlet as well.

**Comment by christianpape** [comment]
Yes, you're right. I'll add the doc section and update the PR...

**Comment by christianpape** [comment]
Docs done!

**Comment by indigo423** [comment]
@christianpape cool thanks!

**Pull request merged**